proto/xmtpv4/message_api/message_api.proto

```diff
@@ -5,20 +5,32 @@ package xmtp.xmtpv4;
 
 import "google/api/annotations.proto";
 import "identity/associations/signature.proto";
+import "mls/api/v1/mls.proto";
 
 option go_package = "github.com/xmtp/proto/v3/go/xmtpv4/message_api";
 
-// Authenticated data within the MlsPrivateMessage
+// Data visible to the server that has been signed by the client.
+// Included in one of:
+// - MLS PrivateMessage.authenticated_data for messages and commits
+// - MLS KeyPackageTBS.extensions for key packages
+// - Omitted for Welcome messages
```

Comment:
I chose to leverage existing fields rather than creating an extra client signature. A drawback is that I can't find an equivalent for Welcome messages, but fortunately causal ordering doesn't make sense for welcome messages (see my comment on the XIP draft).

If we added a standardized signature field for all message types (or just a piecemeal one created for Welcome messages specifically), we could also move the `target_topic` below into the `AuthenticatedData`. If we were to do this, we would need to be careful that the signature is not visible to the server, or we break anonymity.

For now, I assume it's not worth it - the current approach allows us to not modify client interfaces too much, and avoid extra signatures.

Reply:
To access those fields wouldn't the server need to be able to read the TLS serialized data?

Reply:
It's a good point. I don't see an existing library for doing the TLS deserialization in golang, let me know if you find anything. It seems the options are:

1. Expose a callable Rust library to the server for extracting the `AuthenticatedData` from the TLS serialized payload.
2. Add the `AuthenticatedData` as a plaintext field on the `ClientEnvelope`, add a client signature somewhere inside the encrypted payload, and add client logic to validate it after decryption.

(1) is a bit more complex, (2) is more duplicative. Do you have any thoughts here?

Reply:
Summary of discussion: we are going to reuse the validation service (callable Rust library)